code/client/src/state/modules/wallet/reducers.js

```diff
@@ -149,16 +149,31 @@ const reducer = handleActions(
       provider: action.payload,
     }),
 
-    [walletActions.addKnownAddress]: (state, action) => ({
+    [walletActions.setKnownAddress]: (state, action) => ({
       ...state,
       knownAddresses: {
-        ...state.knownAddresses || {},
-        [action.payload.knownAddressKey]: [
-          ...state.knownAddresses?.[action.payload.knownAddressKey] || [],
-          action.payload.oneAddress
-        ]
+        ...state.knownAddresses,
+        [action.payload.address]: {
```

Comment:
guard for undefined `payload.address`?

Reply:
The action payload should be filtered before it reaches the reducer. Trying to keep reducer logic free.